src/examples/common_plane_gpu.cpp

```diff
@@ -0,0 +1,280 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Tribol Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (MIT)
+
+#include "examples_common.hpp" // for common functionality used in examples
+
+#include "tribol/common/ExecModel.hpp"
+#include "tribol/common/Parameters.hpp"
+#include "tribol/config.hpp"
+#include "tribol/interface/tribol.hpp"
+
+#ifdef TRIBOL_USE_UMPIRE
+// Umpire includes
+#include "umpire/ResourceManager.hpp"
+#endif
+
+// Example command line arguments for running this example. This test creates two rectangular blocks of dimensions (l x w x h). The dimensions are 
+// set such that an initial block-intersection exists, triggering the contact interaction. The blocks are discretized per the "block#_res xx yy zz" input 
+// arguments (e.g. block1_res 5 3 4, and block2_res 3 2 1), where "xx", "yy", and "zz" are the number of elements in the x, y and z directions. 
+// Varying "xx" and "yy" will vary the number of contact overlaps that exist between the two surfaces, and therefore, the amount of contact work.
+//
+// srun -n1 ./common_plane_ex --block1_res 100 50 10  --block1_min 0 0 0  --block1_max 10 5 1    --block2_res 150 75 10  --block2_min 0 0 0.95  --block2_max 10 5 1.95
+// srun -n1 ./common_plane_ex --block1_res 100 50 4   --block1_min 0 0 0. --block1_max 1 1 1.05  --block2_res 150 75 4   --block2_min 0 0 0.95  --block2_max 1 1 2
+// srun -n1 ./common_plane_ex --block1_res 4 4 4      --block1_min 0 0 0. --block1_max 1 1 1.05  --block2_res 4 4 4      --block2_min 0 0 0.95  --block2_max 1 1 2
+
+template <tribol::MemorySpace MSPACE, tribol::ExecutionMode EXEC>
+int runExample(const std::string& mesh_file, int num_contact_elems);
+
+/*!
+ * \brief Program main.
+ *
+ * This example runs the common plane + penalty algorithm for two opposing blocks
+ *
+ * \param [in] argc argument counter
+ * \param [in] argv argument vector
+ *
+ * \return rc return code, a non-zero return code indicates an error.
+ */
+int main( int argc, char** argv )
+{
+  ////////////////////////////////
+  //                            //
+  // SETUP THE EXAMPLE AND MESH //
+  //                            //
+  ////////////////////////////////
+
+  // initialize
+#ifdef TRIBOL_USE_MPI
+  MPI_Init( &argc, &argv );
+#endif
+  tribol::CommT problem_comm = TRIBOL_COMM_WORLD;
+  initialize_logger( problem_comm );
+
+#ifdef TRIBOL_USE_UMPIRE
+  umpire::ResourceManager::getInstance();  // initialize umpire's ResouceManager
+#endif
+
+  int err = 0;
+
+  for (int i{0}; i < 11; ++i)
+  {
+    int num_elems_1d = std::pow(2, i);
+    int num_contact_elems = std::pow(num_elems_1d, 2);
+    std::string mesh_file = TRIBOL_REPO_DIR "/data/single_layer_";
```

Comment:
Make a note that this mesh is a 3D mesh? If someone wanted to take this example and modify it, there are parts of the code below (e.g. LINEAR_QUAD in register mesh) that are 3D specific.

Reply:
Mesh is now created inline with `mfem::Mesh::MakeCartesian3D()`